src/client.ts

```diff
@@ -62,8 +62,8 @@ export type RequestConfig = CustomClientOptions & {
  * The Skynet Client which can be used to access Skynet.
  */
 export class SkynetClient {
-  // TODO: This is currently the url of the skapp and not the portal. It should be the value of 'skynet-portal-api' header. This will be a promise, which will be a breaking change.
-  portalUrl: string;
+  // The value of the 'skynet-portal-api' header.
+  portalUrl: Promise<string>;
```

Comment:
This is really unconventional still, is there no way around it? Can we document why it's necessary? I haven't found a clear definition of the difference between an `initialPortalUrl` and an `actualPortalUrl`. I assume it's due to load balancing?